clj/src/cljd/core.cljd

```diff
@@ -7138,13 +7138,72 @@
   [& xforms]
   (Eduction. (apply comp (butlast xforms)) (last xforms) -1))
 
+;; UUID
+(defprotocol IUUID "A marker protocol for UUIDs")
+
+(deftype UUID [uuid ^:mutable __hash]
+  IUUID
+
+  Object
+  (toString [_] uuid)
+
+  IEquiv
+  (-equiv [_ other]
+    (and (satisfies? IUUID other) (identical? uuid (.-uuid other))))
```

Comment:
I think because `toString` already returns the underlying string, we can lean on `str` instead of a specific `-uuid-str` protocol method.

Reply:
Except that if you keep the protocol but use `str` then it may fail when you extend the protocol to, let's say, a 3rd party uuid class where `toString` doesn't return just the UUID string.
That's why to me you have only two choices:
1/ close the implementation to just our UUID class (and still possible to open it up later on when the needs arise)
2/ open the implementation with a proper (≠ marker) protocol

Marker protocols are generally a code smell to me.

Reply:
I went with closing the implementation for now since opening it later makes more sense 👌